platform/common/driver/vault.go

```diff
@@ -112,16 +115,18 @@ type Writes map[Namespace]map[PKey]VaultValue
 
 // VaultLock represents a lock over a transaction or the whole vault
 type VaultLock interface {
+	// Lock acquires the locked resources
+	Lock() error
 	// Release releases the locked resources
 	Release() error
 }
 
 type VaultStore interface {
 	// AcquireTxIDRLock acquires a read lock on a specific transaction.
 	// While holding this lock, other routines:
-	// - cannot update the transaction states or statuses of the locked transactions
-	// - can read the locked transactions
-	AcquireTxIDRLock(ctx context.Context, txID TxID) (VaultLock, error)
+	// - cannot update the transaction states of the locked transactions (based on the isolation level passed)
+	// - can read the locked states
+	AcquireTxIDRLock(ctx context.Context, txID TxID, isolationLevel sql.IsolationLevel) (VaultLock, error)
```

Comment:
It is not clear to me what's the purpose of `AcquireTxIDRLock`. Maybe renaming it to something like `NewTransaction`? How should the developer use this API?